[
  {
    "poster": {
      "poster_name": "Senith Pinsara",
      "poster_handle": "senith"
    },
    "time_stamp": "2023-10-27 01:21:20",
    "text": "`const axios = require(\"axios\");`\n\n`exports.main = async (context = {}, sendResponse) =&gt; {`\n\n  `const { hs_object_id } = context.propertiesToSend;`\n\n  `const response = {`\n    `context: context`\n  `};`\n\n  `try {`\n    `sendResponse(response);`\n  `} catch (error) {`\n    `sendResponse(error);`\n  `}`\n`};`\n\nI'm trying to connect Json based CRM card to React base CRM card.\nMy problem is why `propertiesToSend`  array is null? How to get `hs_object_id`  in this CRM card?",
    "reply_count": 2,
    "reply_users_count": 2,
    "replies": [
      {
        "time_stamp": "2023-10-27 08:48:21"
      },
      {
        "time_stamp": "2023-10-29 21:24:17"
      }
    ]
  },
  {
    "poster": {
      "poster_name": "Yiz Segall",
      "poster_handle": "yiz.segall"
    },
    "time_stamp": "2023-10-27 01:32:25",
    "edited": {
      "time_stamp": "2023-10-27 01:37:29"
    },
    "text": "i'm having issue with the contact finder api:\nI have this api call with HubSpot\n```POST <https://api.hubapi.com/crm/v3/contacts/search>```\n it has the headers 'Content-Type': 'application/json', it has the bear token and has the following data:\n```{ \"filterGroups\": [\n{ \"filters\": [ \n{ \"propertyName\": \"email\", \n\"operator\": \"EQ\", \n\"value\": \"bob@builder.co.uk\" } \n] } \n] } I'm getting a 404 resource not found error even though I am following the documentation and there is such a contact in my crm with that email. The authorization token comes from a private app and is both correct and has the correct scope```",
    "reply_count": 2,
    "reply_users_count": 2,
    "replies": [
      {
        "time_stamp": "2023-10-27 11:53:26"
      },
      {
        "time_stamp": "2023-10-28 13:59:03"
      }
    ]
  },
  {
    "poster": {
      "poster_name": "Luca Bartoli",
      "poster_handle": "bartoli"
    },
    "time_stamp": "2023-10-27 02:03:19",
    "text": "Hi everyone, we had two custom associations (1 to N) between deals and companies, we removed them to create a new one, but it seems the api is not allowing modifying the schema anymore (since yesterday or so)\n`POST <https://api.hubapi.com/crm/v3/schemas/0-2/associations>`\n```{\n\t\"fromObjectTypeId\": \"0-2\",\n\t\"toObjectTypeId\": \"0-3\",\n\t\"name\": \"deal_vendor\",\n\t\"cardinality\": \"ONE_TO_MANY\",\n\t\"inverseCardinality\": \"ONE_TO_ONE\"\n}```\nResponse:\n```{\n\t\"status\": \"error\",\n\t\"message\": \"Cannot modify HubSpot defined object type\",\n\t\"correlationId\": \"4f474cbb-61b0-461a-b318-147348e6c121\",\n\t\"category\": \"VALIDATION_ERROR\",\n\t\"subCategory\": \"ObjectSchemaError.CANNOT_MODIFY_HubSpot_DEFINED_OBJECT_TYPE\"\n}```\nThe same still works if one of the two objects is a custom object.\n\nAny help?",
    "reply_count": 8,
    "reply_users_count": 2,
    "replies": [
      {
        "time_stamp": "2023-10-27 05:33:23"
      },
      {
        "time_stamp": "2023-10-27 05:33:37"
      },
      {
        "time_stamp": "2023-10-27 05:34:02"
      },
      {
        "time_stamp": "2023-10-27 05:34:13"
      },
      {
        "time_stamp": "2023-10-27 05:34:41"
      },
      {
        "time_stamp": "2023-10-27 05:41:10"
      },
      {
        "time_stamp": "2023-10-27 05:43:59"
      },
      {
        "time_stamp": "2023-10-27 06:10:18"
      }
    ]
  },
  {
    "poster": {
      "poster_name": "Luca Bartoli",
      "poster_handle": "bartoli"
    },
    "time_stamp": "2023-10-27 02:07:11",
    "text": "it seems like you're missing the association type:\n\n```\"associations\": [\n    {\n      \"to\": {\n        \"id\": \"136551\"\n      },\n      \"types\": [\n        {\n          \"associationCategory\": \"USER_DEFINED\", //Or HubSpot_DEFINED\n          \"associationTypeId\": 2 //Or whatever is your type id\n        }\n      ]\n    }\n  ],```"
  },
  {
    "poster": {
      "poster_name": "Thomas Balsløv",
      "poster_handle": "tba"
    },
    "time_stamp": "2023-10-27 02:35:38",
    "text": "Hi all.\n\"All contacts\" API endpoint not returning all contacts\nWe are using a sync job to migrate data from our production database to HubSpot by extracting all contacts and updating them back into HubSpot.\nWe are however not seeing all the contacts returned via the API (<https://legacydocs.HubSpot.com/docs/methods/contacts/get_contacts><https://legacydocs.HubSpot.com/docs/methods/contacts/get_contacts),|),> only half are returned to us. We are using a simple loop going over each returned page (of 100 items) until the \"has-more\" flag turns false. We are not seeing any errors from the API (only 200 OKs).\n\nThe issue is quite critical to us as the client status is not propagated to HubSpot for customer handling and care.\n\nI would like help in resolving this issue.\n\nBest regards, Thomas.\nDeveloper at Mouseflow.",
    "reply_count": 1,
    "reply_users_count": 1,
    "replies": [
      {
        "time_stamp": "2023-10-27 10:28:19"
      }
    ]
  },
  {
    "poster": {
      "poster_name": "Bob Balm",
      "poster_handle": "bob209"
    },
    "time_stamp": "2023-10-27 05:33:23",
    "text": "Try using */crm/v4/associations/{fromObjectType}/{toObjectType}/labels*"
  },
  {
    "poster": {
      "poster_name": "Bob Balm",
      "poster_handle": "bob209"
    },
    "time_stamp": "2023-10-27 05:33:37",
    "text": "I was able to create a Vendor label for Companies &lt;&gt; Deals no problem"
  },
  {
    "poster": {
      "poster_name": "Bob Balm",
      "poster_handle": "bob209"
    },
    "time_stamp": "2023-10-27 05:34:02",
    "text": "it doesn't allow you to set cardinality though"
  },
  {
    "poster": {
      "poster_name": "Luca Bartoli",
      "poster_handle": "bartoli"
    },
    "time_stamp": "2023-10-27 05:34:13",
    "text": "Of course, that's the entire point"
  },
  {
    "poster": {
      "poster_name": "Luca Bartoli",
      "poster_handle": "bartoli"
    },
    "time_stamp": "2023-10-27 05:34:41",
    "text": "V3 is the only endpoint allowing to set a 1toN cardinality as far as I know"
  },
  {
    "poster": {
      "poster_name": "Bob Balm",
      "poster_handle": "bob209"
    },
    "time_stamp": "2023-10-27 05:41:10",
    "text": "I did notice that v4 documentation doesn't mention cardinality anymore and also in the HubSpot interface, many-to-many is the only option you can choose. Seems like this is something others have found too: <https://community.HubSpot.com/t5/APIs-Integrations/One-to-many-instead-of-many-to-many/td-p/847571>"
  },
  {
    "poster": {
      "poster_name": "Bob Balm",
      "poster_handle": "bob209"
    },
    "time_stamp": "2023-10-27 05:43:59",
    "text": "And this: <https://community.HubSpot.com/t5/HubSpot-Ideas/Possibility-of-1-1-or-1-M-associations-between-all-objects/idi-p/626938>\n\nSomeone there said: \"Did HubSpot never think that there would be scenarios where its users would need to avoid M:M relationships?\" - that's been my suspicion as well. Either that or it's still on the backlog."
  },
  {
    "poster": {
      "poster_name": "Luca Bartoli",
      "poster_handle": "bartoli"
    },
    "time_stamp": "2023-10-27 06:10:18",
    "text": "Well, we have an internal tool which allows admins to generate 1toN and NtoM without issues on custom objects. Under the hood, it uses the v3.\nI'm pretty sure the same endpoint used to work until some days ago.\nHubSpot announced earlier this year that they would have rolled out a way to manage associations limits on objects. The announcement post is still silent, but it was scheduled for development on Sept/Oct 2023. I suppose this can be related to HubSpot objects not working anymore on the v3 endpoint"
  },
  {
    "poster": {
      "poster_name": "Mubariz Muneer",
      "poster_handle": "mubariz.muneer"
    },
    "time_stamp": "2023-10-27 06:48:25",
    "text": "<@U03REFNRMB3> Hey thanks for the reply, I want to use it for external website."
  },
  {
    "poster": {
      "poster_name": "Mukesh Pal",
      "poster_handle": "mukesh.pal"
    },
    "time_stamp": "2023-10-27 07:12:19",
    "text": "```for associationCategory I passed HubSpot_DEFINED and for associationTypeId I tried with 0,1,2,3\nBelow error I'm getting```\n{\n    \"status\": \"error\",\n    \"message\": \"invalid from object type 0-3 for associations to be created. expected: 0-1\",\n    \"correlationId\": \"cd348db7-8874-45a5-9820-9542d789b3e6\",\n    \"category\": \"VALIDATION_ERROR\"\n}"
  },
  {
    "poster": {
      "poster_name": "Mukesh Pal",
      "poster_handle": "mukesh.pal"
    },
    "time_stamp": "2023-10-27 07:18:00",
    "text": "Hi,\nI have a question regarding api integration, I have to associate Contacts, Deals and Line of Items.\nFirst I run the Contacts create api, record get insert and reflected on HubSpot\nNow I want to run Deals create api, in this what parameter should I pass to associate with Contact?\n\nI have tried to pass the id (response get from contact api) in deal api, it run successfully but on HubSpot dashboard it shows zero deal in that contact.\nSimilarly I tried with Line of Item api and it shows zero line of item in deal.\nPlease help me to make these things workable."
  },
  {
    "poster": {
      "poster_name": "Niko Dixon",
      "poster_handle": "niko209"
    },
    "time_stamp": "2023-10-27 08:48:21",
    "text": "have you added the object id to the propertiesToSend array in the  card’s .json config file?"
  },
  {
    "poster": {
      "poster_name": "Eric Goldman",
      "poster_handle": "eric732"
    },
    "time_stamp": "2023-10-27 10:28:19",
    "text": "Hey <@U0635H40MBN> — your note caught my eye. I think this is a problem our product solves well for devs → <http://sequin.io|sequin.io>\n\nI’m biased, but wanted to put it on your radar."
  },
  {
    "poster": {
      "poster_name": "Justin Givens - Level 2",
      "poster_handle": "jgivens"
    },
    "time_stamp": "2023-10-27 10:44:29",
    "text": "Cool! check out this example of who you can do with a simple APP and use a single scope.\n\n<https://plugins.miniorange.com/configure-HubSpot-sign-in-for-wordpress-website>"
  },
  {
    "poster": {
      "poster_name": "Carter McKay - Level 17",
      "poster_handle": "carter"
    },
    "time_stamp": "2023-10-27 11:53:26",
    "text": "The url should be <https://api.hubapi.com/crm/v3/objects/contacts/search>"
  },
  {
    "poster": {
      "poster_name": "Hollyce Brammer",
      "poster_handle": "hollyce"
    },
    "time_stamp": "2023-10-27 12:36:20",
    "text": "<@U04HSBLHAJZ> Hi Ben. Did you ever find a solution for this? I need the same solution for a client."
  },
  {
    "poster": {
      "poster_name": "Ben Lerner",
      "poster_handle": "ben198"
    },
    "time_stamp": "2023-10-27 12:38:02",
    "text": "Afraid after much research and confirmation from the HubSpot team it was not possible.   It might be possible now, but doubt it as I was told it wasnt even on the roadmap."
  },
  {
    "poster": {
      "poster_name": "Ben Lerner",
      "poster_handle": "ben198"
    },
    "time_stamp": "2023-10-27 12:38:03",
    "text": "Sorry!"
  },
  {
    "poster": {
      "poster_name": "Hollyce Brammer",
      "poster_handle": "hollyce"
    },
    "time_stamp": "2023-10-27 12:38:18",
    "text": "Thanks a ton!"
  },
  {
    "poster": {
      "poster_name": "hub.sketch",
      "poster_handle": "hub.sketch"
    },
    "time_stamp": "2023-10-27 12:46:21",
    "text": "<@U03REFNRMB3> I checked the property history and it does not show the data itself, it shows that a change happened and who did it but that's about it. do you have a snapshot you can share where it shows the data in the history?"
  },
  {
    "poster": {
      "poster_name": "Justin Givens - Level 2",
      "poster_handle": "jgivens"
    },
    "time_stamp": "2023-10-27 12:47:44",
    "text": "Yes, that is correct.\n\nFor example, if the contact firstname was “John” and then your API updated it to blank (null). Then the property details (history) would says API App change it from John to Blank."
  }
]